x-pack/legacy/plugins/siem/server/lib/note/saved_object.ts

```diff
@@ -31,20 +29,11 @@ import { pickSavedTimeline } from '../timeline/pick_saved_timeline';
 import { convertSavedObjectToSavedTimeline } from '../timeline/convert_saved_object_to_savedtimeline';
 
 export class Note {
-  constructor(
-    private readonly libs: {
-      savedObjects: Pick<Legacy.SavedObjectsService, 'getScopedSavedObjectsClient'> &
-        Pick3<Legacy.SavedObjectsService, 'SavedObjectsClient', 'errors', 'isConflictError'>;
-    }
-  ) {}
-
   public async deleteNote(request: FrameworkRequest, noteIds: string[]) {
+    const savedObjectsClient = request.context.core.savedObjects.client;
```

Comment:
@rudolf is the SavedObjects client going to be available as a top-level plugin, or is it only accessible through the router's context? I see that e.g. IndexPatternsFetcher is the former, while for SavedObjects it's currently the latter.

Our framework abstraction used to sit between our client usage and the Hapi.Server object, and it would be nice to adapt that to work with the new platform in a request-agnostic manner.

P.S. the fact that the context here is on a request will be addressed in #52133.

Reply:
Whoops, I see now that the client is/was scoped based on the request, and that the start contract provides a more analogous method for retrieving a scoped client based on a request. I'll figure out where we want to land on this in #52133.